ptsprojects/boards/__init__.py

```diff
@@ -118,14 +119,20 @@ def get_device_list():
     return device_list
 
 
-def get_free_device():
+def get_free_device(board=None):
     """Returns tty path and jlink serial number of a free device."""
     devices = get_device_list()
 
-    for dev in devices.keys():
-        if dev not in devices_in_use:
-            devices_in_use.append(dev)
-            return dev, devices[dev]
+    SNR_PREFIX_FOR_BOARD = {
```

Comment:
Maybe a defaultdict could be used here so when there is an unknown board then an empty string would be returned?

Reply:
Wouldn't that mean that an unknown `board` argument would be ignored? I would rather it fail loudly if I pass an unknown board argument.

Reply:
The thing is I don't want to break this for other boards like reel_board. Anyway, why do we need this prefix check?

Reply:
I hadn't considered that `"reel_board"` may also be given as an argument here. Will switch to a `defaultdict(str)`.

The prefix check is needed if one has both 52 and 53 boards connected. Without the prefix check a 52 board may be selected even if `board` is `"nrf53"`.